java/source/dataset.rst

```diff
@@ -0,0 +1,300 @@
+.. _arrow-dataset:
+
+=======
+Dataset
+=======
+
+* `Arrow Java Dataset <https://arrow.apache.org/docs/dev/java/dataset.html>`_: Java implementation of Arrow Datasets library.
```

Comment:
Maybe make it explicit that it's JNI bindings, not a separate implementation?

Can we use Intersphinx to do this linking? (Example for Python: https://github.com/apache/arrow-cookbook/blob/main/python/source/flight.rst#authentication-with-userpassword)

Reply:
I didn't see a change here, did I miss it?

Reply:
Sorry, were problems in pull/merge, added finally